### PR TITLE
Remove NonDebuggable from Azure Blob Storage

### LINKED ISSUE #875: [BC Idea]: Remove NonDebuggable property from several parts of the system application
### BC Idea Link

https://experience.dynamics.com/ideas/idea/?ideaid=5c138a47-a3f2-ee11-a73d-6045bdb71da1

### Description

Currently there are still a lot of procedures that doesn't allow debugging.

This might make sense for Authorization and user based information, but sometimes the NonDebuggable property seems to forgotten and should be removed.

Since it's difficult to do this by my own without Microsoft beeing involed I'd like to discuss the details before removing the NonDebuggable property.

here some example

```al
//codeunit 9350 "Graph Client"
    var
        [NonDebuggable] //Makes it impossible to debug the implementation
        GraphClientImpl: Codeunit "Graph Client Impl.";
```

```al
//codeunit 7803 "Azure Functions Impl"
    [NonDebuggable]
    procedure SendGetRequest(AzureFunctionAuthentication: Interface "Azure Functions Authentication"; QueryDict: Dictionary of [Text, Text]): Codeunit "Azure Functions Response"
    begin
        exit(Send(AzureFunctionAuthentication, Enum::"Http Request Type"::GET, QueryDict, '', ''));
    end;
```

### I will provide the implementation for this BC Idea

- [X] I will provide the implementation for this BC Idea
Internal work item: [AB#524277](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/524277)

### PR DESCRIPTION
**Pull request by pri-kise**
#### Summary
Remove NonDebuggable from Azure Blob Storage and Storage Authorization

This PR removes the NonDebuggable from Helper procedure and other procedures where the data of the procedures is already known, since it's provided from upper codeunits.

I skipped the RequestHelper Codeunit for the moment, since I'm not sure if we can remove the NonDebuggable property there.
e.g. if we use the ReadSAS Authorization where the Secret is part of the request uri.

I added new Overloads to the `codeunit 9062 "Storage Service Authorization"` for `CreateAccountSAS` with SecretText.

#### Work Item(s)
Partially Fixes #875

Fixes [AB#524277](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/524277)

### CODE REVIEW

File: src/System Application/App/Azure Blob Services API/src/ABSOperationPayload.Codeunit.al [hunk 1]
```diff
@@ -14,22 +14,15 @@ codeunit 9042 "ABS Operation Payload"
     InherentPermissions = X;
 
     var
-        [NonDebuggable]
         ContentHeaders: Dictionary of [Text, Text];
-        [NonDebuggable]
         RequestHeaders: Dictionary of [Text, Text];
-        [NonDebuggable]
         UriParameters: Dictionary of [Text, Text];
```
Comment (darjoo) [inline src/System Application/App/Azure Blob Services API/src/ABSOperationPayload.Codeunit.al]:
```suggestion
```

**Review by darjoo — APPROVED** [review]

**Review by darjoo — DISMISSED** [review]

**Review by JesperSchulz — APPROVED** [review]

**Comment by JesperSchulz** [comment]
@pri-kise, you seem to have compilation issues:
Error: AL0432 Method 'CreateAccountSAS' is marked for removal. Reason: Use CreateAccountSAS with SecretText data type for SigningKey.. Tag: 25.0.

**Comment by pri-kise** [comment]
@JesperSchulz I've updated the test to use SecretKey
Hopefull the CI pass now.

**Review by JesperSchulz — APPROVED** [review]

File: src/System Application/App/Azure Storage Services Authorization/src/StorageServiceAuthorization.Codeunit.al [hunk 2]
```diff
@@ -54,6 +57,46 @@ codeunit 9062 "Storage Service Authorization"
...
+    /// </summary>
+    /// <param name="SigningKey">The signing key to use.</param>
+    /// <param name="SignedVersion">Specifies the signed storage service version to use to authorize requests made with this account SAS. Must be set to version 2015-04-05 or later.</param>
+    /// <param name="SignedServices">Specifies the signed services accessible with the account SAS.</param>
+    /// <param name="SignedPermissions">Specifies the signed permissions for the account SAS. Permissions are only valid if they match the specified signed resource type; otherwise they are ignored.</param>
+    /// <param name="SignedExpiry">The time at which the shared access signature becomes invalid.</param>
+    /// <param name="OptionalSASParameters">See table "Stor. Serv. SAS Parameters".</param>
+    /// <returns>An account SAS authorization.</returns>
+    procedure CreateAccountSAS(SigningKey: SecretText; SignedVersion: Enum "Storage Service API Version"; SignedServices: List of [Enum "SAS Service Type"];
+                                                                    SignedResources: List of [Enum "SAS Resource Type"];
```
Comment (darjoo) [inline src/System Application/App/Azure Storage Services Authorization/src/StorageServiceAuthorization.Codeunit.al]: I don't want to nag.. But this bothers me :D Having some variables on new lines.

**Review by darjoo — APPROVED** [review]

File: src/System Application/App/Azure Storage Services Authorization/src/StorageServiceAuthorization.Codeunit.al [hunk 3]
```diff
@@ -54,6 +57,46 @@ codeunit 9062 "Storage Service Authorization"
...
+                                                                    SignedPermissions: List of [Enum "SAS Permission"];
+                                                                    SignedExpiry: DateTime;
+                                                                    OptionalSASParameters: Record "SAS Parameters"): Interface "Storage Service Authorization"
```
Comment (JesperSchulz) [inline src/System Application/App/Azure Storage Services Authorization/src/StorageServiceAuthorization.Codeunit.al]:
```suggestion
    procedure CreateAccountSAS(SigningKey: SecretText; SignedVersion: Enum "Storage Service API Version"; SignedServices: List of [Enum "SAS Service Type"]; SignedResources: List of [Enum "SAS Resource Type"]; SignedPermissions: List of [Enum "SAS Permission"]; SignedExpiry: DateTime; OptionalSASParameters: Record "SAS Parameters"): Interface "Storage Service Authorization"
```

**Review by darjoo — APPROVED** [review]

File: src/System Application/App/Azure Storage Services Authorization/src/StorageServiceAuthorization.Codeunit.al (hunk 2 above)
Comment (JesperSchulz) [reply src/System Application/App/Azure Storage Services Authorization/src/StorageServiceAuthorization.Codeunit.al]: I think I managed to resolve that one :) Let's run one more time.

**Review by JesperSchulz — APPROVED** [review]

**Pull request merged**